### PR TITLE
drivers: lora: Clean up SX126x pin handling

### PR DESCRIPTION
**Pull request by andysan**
Port the PIN configuration helpers from the SX1276 driver to the SX126x driver.

### CODE REVIEW

File: drivers/lora/sx12xx_common.h [hunk 1]
```diff
@@ -11,6 +12,18 @@
 #include <drivers/lora.h>
 #include <device.h>
 
+int __sx12xx_configure_pin(struct device **dev, const char *controller,
+			   gpio_pin_t pin, gpio_flags_t flags);
```
Comment (mniestroj) [inline drivers/lora/sx12xx_common.h:16]: I think `#include <drivers/gpio.h>` should be in this header file, so `gpio_pin_t` and `gpio_flags_t` are known.

File: drivers/lora/sx126x.c [hunk 2]
```diff
@@ -381,34 +381,13 @@ static int sx126x_lora_init(struct device *dev)
 
 	LOG_DBG("Initializing %s", DT_INST_LABEL(0));
 
-	dev_data.reset = device_get_binding(GPIO_RESET_LABEL);
-	if (!dev_data.reset) {
-		LOG_ERR("Cannot get pointer to %s device", GPIO_RESET_LABEL);
+	if (sx12xx_configure_pin(reset, GPIO_OUTPUT_ACTIVE) ||
```
Comment (mniestroj) [inline drivers/lora/sx126x.c:384]: Macros no longer used (to be removed):
- GPIO_RESET_LABEL
- GPIO_RESET_FLAGS
- GPIO_BUSY_LABEL
- GPIO_BUSY_FLAGS
- GPIO_DIO1_LABEL
- GPIO_DIO1_FLAGS
- GPIO_ANTENNA_ENABLE_LABEL
- GPIO_ANTENNA_ENABLE_FLAGS

**Review by mniestroj — CHANGES_REQUESTED** [review]
Overall looks good, especially code reduction in last commit :) See my comments.

File: drivers/lora/sx126x.c (hunk 2 above)
Comment (andysan) [reply drivers/lora/sx126x.c:371]: Thanks! I thought I did that already. I have removed these from the updated commit.

File: drivers/lora/sx12xx_common.h (hunk 1 above)
Comment (Mani-Sadhasivam) [reply drivers/lora/sx12xx_common.h:16]: > I think `#include <drivers/gpio.h>` should be in this header file, so `gpio_pin_t` and `gpio_flags_t` are known.

Ack

**Review by Mani-Sadhasivam — APPROVED** [review]
Except the comment from @mniestroj, rest LGTM!

File: drivers/lora/sx12xx_common.h (hunk 1 above)
Comment (andysan) [reply drivers/lora/sx12xx_common.h:17]: Ugh, I seem to have missed this comment for some reason. I have updated the commit. Thanks for spotting it!

**Review by mniestroj — APPROVED** [review]
LGTM

**Pull request merged**